datatypes/src/operations/image/colorizer.rs

```diff
@@ -458,28 +585,33 @@ impl<'c> ColorMapper<'c> {
                 min_value,
                 max_value,
                 no_data_color,
-                default_color,
+                over_color,
+                under_color,
             } => {
                 let value: f64 = value.as_();
                 if f64::is_nan(value) {
                     *no_data_color
-                } else if value < *min_value || value > *max_value {
-                    *default_color
+                } else if value < *min_value {
+                    *under_color
+                } else if value > *max_value {
+                    *over_color
                 } else {
                     let color_table_factor = (color_table.len() - 1) as f64;
                     let table_entry = f64::round(
                         color_table_factor * ((value - *min_value) / (*max_value - *min_value)),
                     ) as usize;
-                    *color_table.get(table_entry).unwrap_or(default_color)
+                    // TODO:                                    vvv ----this was default color, what should it be now?
+                    *color_table.get(table_entry).unwrap_or(no_data_color)
```

Comment:
cf. comment above

Reply:
But in the case of a color table, its used only with linear or log colorizer, isn't it? And there we don't have the default color anymore now. Should it be transparent in this case?

Reply:
This case should not happen in general.
I guess, the most reasonable thing would be to check if `table_entry` would be < 0 and then use the under_color.

If the entry is above the largest index, we should use the over color.

You can check if the first case can ever happen. If not, we should use the over color.

Reply:
Ok, upon further investigation, I think we could go with an `except(...)` call here (or add a new error?), because I don't see any possibility for this call to fail.

```
let color_table_factor = (color_table.len() - 1) as f64;
let table_entry = f64::round(
    color_table_factor * ((value - *min_value) / (*max_value - *min_value))
) as usize;
```
There is no way for this to be outside of the index set of the color table, unless `value` is outside of the min/max range. But in that case, the other conditional branches would have caught it.

Reply:
In theory, yes, but the round computation is done within this branch. This does not seem super robust.